### PR TITLE
samples: added cleanup option

### LINKED ISSUE #3836: com.example.cloud.iot.endtoend.CloudiotPubsubExampleServerTest: testConfigTurnOn failed
This test failed!

To configure my behavior, see [the Build Cop Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/buildcop).

If I'm commenting on this issue too often, add the `buildcop: quiet` label and
I will stop commenting.

---

commit: c2497c227b59186e603ade8179d2427f63e0241f
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/12fa1558-b097-4642-8111-f120eeef1ff4), [Sponge](http://sponge2/12fa1558-b097-4642-8111-f120eeef1ff4)
status: failed
<details><summary>Test output</summary><br><pre>com.google.api.client.googleapis.json.GoogleJsonResponseException: 
429 Too Many Requests
POST https://cloudiot.googleapis.com/v1/projects/java-docs-samples-testing/locations/us-central1/registries
{
  "code" : 429,
  "errors" : [ {
    "domain" : "global",
    "message" : "Maximum number of registries resources exceeded. Couldn't create registry.",
    "reason" : "rateLimitExceeded"
  } ],
  "message" : "Maximum number of registries resources exceeded. Couldn't create registry.",
  "status" : "RESOURCE_EXHAUSTED"
}
	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:149)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:112)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:39)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:443)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1108)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:541)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:474)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:591)
	at com.example.cloud.iot.endtoend.CloudiotPubsubExampleServer.createRegistry(CloudiotPubsubExampleServer.java:112)
	at com.example.cloud.iot.endtoend.CloudiotPubsubExampleServerTest.testConfigTurnOn(CloudiotPubsubExampleServerTest.java:67)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
</pre></details>

### PR DESCRIPTION
**Pull request by munkhuushmgl**
Fixes #3836 

- [X] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: iot/api-client/end-to-end-example/src/test/java/com/example/cloud/iot/endtoend/CloudiotPubsubExampleServerTest.java [hunk 1]
```diff
@@ -80,6 +95,37 @@ public void testConfigTurnOn() throws GeneralSecurityException, IOException, JSO
...
+      }
+    }
+
+    // Delete the 50 oldest registries
+    for (DeviceRegistry registry : filteredRegistries) {
+      CleanUpHelper.clearRegistry(CLOUD_REGION, PROJECT_ID, registry.getId());
+    }
+    //      if()
+    //      CleanUpHelper.clearRegistry();
+    //    }
```
Comment (lesv) [inline iot/api-client/end-to-end-example/src/test/java/com/example/cloud/iot/endtoend/CloudiotPubsubExampleServerTest.java:126]: Do you mean for this to be here?

**Review by lesv — APPROVED** [review]
One open question; Ok to merge if it's intentional.  Please fix if not, and ping me.

File: iot/api-client/end-to-end-example/src/test/java/com/example/cloud/iot/endtoend/CloudiotPubsubExampleServerTest.java (hunk 1 above)
Comment (munkhuushmgl) [reply iot/api-client/end-to-end-example/src/test/java/com/example/cloud/iot/endtoend/CloudiotPubsubExampleServerTest.java:126]: oops
I will fix that

**Comment by munkhuushmgl** [comment]
> One open question; Ok to merge if it's intentional. Please fix if not, and ping me.
PLease merge it after the fix

**Pull request merged**